packages/providers/src/provider.ts

```diff
@@ -209,15 +211,34 @@ export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
 
+  consensusParamsCache?: ConsensusParameters;
+
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
     public options: ProviderOptions = {}
   ) {
+    // TODO: Disable or discourage the usage of this constructor in favour of connect()
     this.operations = this.createOperations(url, options);
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const provider = new Provider(url, options);
+    const { consensusParameters } = await provider.getChain();
+    provider.consensusParamsCache = consensusParameters;
+    return provider;
+  }
```

Comment:
Instead of telling users to create providers using `new Provider()`, we will now encourage users to run `await Provider.connect('url')` instead.

This new `connect` method calls the constructor and does some additional work afterwards. It fetches the consensus params from the chain, and simply stores them on the `Provider` object for later use.

Reply:
It would make sense to make the constructor private and force users to only instantiate the Provider class by `await Provider.connect('url')`. This way we make sure that the `consensusParams` is always going to be available within a Provider instance